src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -2003,11 +2004,12 @@ protected RubyString scrubAsciiCompat(Object string, RubyProc block,
             var tstring = strings.getTString(string);
             TruffleString buf = EMPTY_BINARY_TSTRING;
 
-            final byte[] pBytes = bytesNode.execute(rope);
-            final int e = pBytes.length;
+            var byteArray = byteArrayNode.execute(tstring, tencoding);
+            final byte[] pBytes = byteArray.getArray();
+            final int e = tstring.byteLength(tencoding);
```

Comment:
I think `e` should be `byteArray.getEnd()` here, it looks like an "end" not a "length" together with `p`.
It's already getEnd() in the specialization below.

Reply:
This turned out to be problematic in other ways. I needed to subtract back out the offset for the substring operation. See 59692a5bc63b5be87ca9a19578a04aecf24f7c7b.